src/Hosting/Hosting/src/Internal/HostingEventSource.cs

```diff
@@ -19,6 +19,10 @@ internal sealed class HostingEventSource : EventSource
     private long _currentRequests;
     private long _failedRequests;
 
+    public long TotalRequests => Volatile.Read(ref _totalRequests);
+    public long CurrentRequests => Volatile.Read(ref _currentRequests);
+    public long FailedRequests => Volatile.Read(ref _failedRequests);
```

Comment:
I left `HostingEventSource` essentially unchanged. It's the source of truth of the counters. It seemed the simplest solution where we have already existing event source counters.